liquid/commitment.go

```diff
@@ -0,0 +1,131 @@
+	Commitments []Commitment `json:"commitments"`
+}
+
+// Commitment appears in type CommitmentChangeRequest.
+//
+// The commitment is located in a certain project and applies to a certain resource within a certain AZ.
+// These metadata are implied by where the commitment is found within type CommitmentChangeRequest.
+type Commitment struct {
+	// The same UUID may appear multiple times within the same changeset for one specific circumstance:
+	// If a commitment moves between projects, it will appear as being deleted in the source project and again as being created in the target project.
+	UUID string `json:"uuid"`
+
+	// These two status fields communicate one of three possibilities:
+	//   - If OldStatus.IsNone() and NewStatus.IsSome(), the commitment is being created (or moved to this location).
+	//   - If OldStatus.IsSome() and NewStatus.IsNone(), the commitment is being deleted (or moved away from this location).
+	//   - If OldStatus.IsSome() and NewStatus.IsSome(), the commitment is only changing its status (e.g. from "active" to "expired" when ExpiresAt has passed).
+	OldStatus Option[CommitmentStatus] `json:"oldStatus"`
+	NewStatus Option[CommitmentStatus] `json:"newStatus"`
+
+	Amount uint64 `json:"amount"`
+
+	// For commitments in status "planned", this field contains the point in time in the future when the user wants for it to move into status "confirmed".
+	// If confirmation is not possible by that point in time, the commitment will move into status "pending" until it can be confirmed.
+	//
+	// For all other status values, this field contains the point in time when the status transitioned from "active" to "pending",
+	// or None() if the commitment was created for immediate confirmation and therefore started in status "pending".
+	ConfirmBy Option[time.Time] `json:"confirmBy,omitzero"`
+
+	// This field contains the point in time when the commitment moves into status "expired", unless it is deleted or moves into status "superseded" first.
+	ExpiresAt time.Time `json:"expiresAt,omitzero"`
+}
+
+// CommitmentStatus is an enum containing the various lifecycle states of type Commitment.
+type CommitmentStatus string
+
+const (
+	// CommitmentStatusPlanned means that the commitment has a ConfirmBy date in the future.
+	// Planned commitments are used to notify the cloud about future resource demand.
+	CommitmentStatusPlanned CommitmentStatus = "planned"
+	// CommitmentStatusPending means that the commitment has a ConfirmBy date in the past, but the cloud has not confirmed it yet.
+	// Pending commitments usually occur when there is not enough capacity to cover all current resource demands.
+	CommitmentStatusPending CommitmentStatus = "pending"
+	// CommitmentStatusConfirmed means that the commitment has been confirmed and is being honored by the cloud.
+	// Confirmed commitments represent current resource demand that the cloud is able to guarantee.
```

Comment:
I suggest to include commitments with future time explicitly. 
```suggestion
	// CommitmentStatusConfirmed means that the commitment has been confirmed and is being honored by the cloud.
	// Confirmed commitments represent resource demand that the cloud is able to guarantee, either immediately or at a specified future time (as indicated by the ConfirmBy field).
```

Reply:
Hm, but then I miss something to differentiate commitments with a future ConfirmBy date that are either A) not confirmed yet, or B) confirmed.

Reply:
result of live discussion: we will add a new state: `guaranteed` for commitments with a future ConfirmBy date that are confirmed. 
So commitments start either in state `planned` or `guaranteed`.
"positive" transitions are `planned` -->  `pending` -->  `confirmed` and `guaranteed` -->  `confirmed`